### PR TITLE
Fix CodeQL errors

### PR DESCRIPTION
**Pull request by kenlautner**
## Description

Fixed some CodeQL failures we're seeing in a variety of packages.  This takes the changes from the PR found [here](https://github.com/microsoft/mu_basecore/pull/488) and builds upon them.

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [x] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

Tested through CodeQL checks.

## Integration Instructions

N/A

### CODE REVIEW

File: MdeModulePkg/Core/Pei/Ppi/Ppi.c [hunk 1]
```diff
@@ -324,16 +324,9 @@ ConvertPpiPointersFv (
...
-      // Instead we compare the GUID as INT32 at a time and branch
-      // on the first failed comparison.
-      //
-      if ((((INT32 *)Guid)[0] == ((INT32 *)GuidCheckList[GuidIndex])[0]) &&
-          (((INT32 *)Guid)[1] == ((INT32 *)GuidCheckList[GuidIndex])[1]) &&
-          (((INT32 *)Guid)[2] == ((INT32 *)GuidCheckList[GuidIndex])[2]) &&
-          (((INT32 *)Guid)[3] == ((INT32 *)GuidCheckList[GuidIndex])[3]))
-      {
+      // MU_CHANGE [BEGIN] - CodeQL change
+      if (CompareMem (Guid, GuidCheckList[GuidIndex], 16) == 0) {
```
Comment (makubacki) [inline MdeModulePkg/Core/Pei/Ppi/Ppi.c]: Like the other PR, I'd use `CompareGuid()` here if there's not a known reason to avoid it.

Comment (kenlautner) [reply MdeModulePkg/Core/Pei/Ppi/Ppi.c]: No reason so I changed it.

File: CryptoPkg/Test/UnitTest/Library/BaseCryptLib/X509Tests.c [hunk 2]
```diff
@@ -573,7 +573,14 @@ TestVerifyX509 (
   Status      = X509GetIssuerName (mTestCert, sizeof (mTestCert), NULL, &SubjectSize);
   UT_ASSERT_TRUE (!Status);
   Subject = AllocatePool (SubjectSize);
-  Status  = X509GetIssuerName (mTestCert, sizeof (mTestCert), Subject, &SubjectSize);
+  // MU_CHANGE [BEGIN] - CodeQL change
```
Comment (kuqin12) [inline CryptoPkg/Test/UnitTest/Library/BaseCryptLib/X509Tests.c:576]: Does this happen in 2208 as well?

**Review by kuqin12 — APPROVED** [review]

File: CryptoPkg/Test/UnitTest/Library/BaseCryptLib/X509Tests.c (hunk 2 above)
Comment (kenlautner) [reply CryptoPkg/Test/UnitTest/Library/BaseCryptLib/X509Tests.c:576]: No.  This test was added with 202302.

**Review by makubacki — APPROVED** [review]

**Pull request merged**